src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClone = [];
+  const oneState = {};
+
+  Object.assign(oneState, state);
+
+  for (const step in actions) {
+    switch (actions[step].type) {
+      case 'addProperties':
+        Object.assign(oneState, actions[step].extraData);
+        break;
+      case 'removeProperties':
+        for (const key in actions[step].keysToRemove) {
+          const removeKey = actions[step].keysToRemove[key];
+
+          delete oneState[removeKey];
+        }
+        break;
+
+      case 'clear':
+        for (const key in oneState) {
+          delete oneState[key];
+        }
+    }
+
+    stateClone.push(Object.assign({}, oneState));
```

Comment:
According to the additional prompt rules, it is better to push a copy of the object to the array at the end of each loop cycle, but outside of the `switch` block. Consider moving this line outside the `switch` block, after the current action is processed.